packages/alchemy/src/__tests__/provider.test.ts

```diff
@@ -66,6 +66,166 @@ describe("Alchemy Provider Tests", () => {
       '"0x18dfb3c7000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000002000000000000000000000000deadbeefdeadbeefdeadbeefdeadbeefdeadbeef0000000000000000000000008ba1f109551bd432803012645ac136ddd64dba720000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000004deadbeef000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000004cafebabe00000000000000000000000000000000000000000000000000000000"'
     );
   });
+
+  it("should correctly do runtime validation when entrypoint is invalid", () => {
+    expect(
+      () =>
+        new AlchemyProvider({
+          rpcUrl: "https://eth-mainnet.g.alchemy.com/v2/test",
+          entryPointAddress: 1 as unknown as Address,
+          chain: polygonMumbai,
+        })
+    ).toThrowErrorMatchingInlineSnapshot(`
+      "[
+        {
+          \\"code\\": \\"invalid_type\\",
+          \\"expected\\": \\"string\\",
+          \\"received\\": \\"number\\",
+          \\"path\\": [
+            \\"entryPointAddress\\"
+          ],
+          \\"message\\": \\"Expected string, received number\\"
+        }
+      ]"
+    `);
+  });
+
+  it("should correctly do runtime validation when multiple inputs are invalid", () => {
+    expect(
+      () =>
+        new AlchemyProvider({
+          rpcUrl: 1 as unknown as string,
+          entryPointAddress: 1 as unknown as Address,
+          chain: polygonMumbai,
+        })
+    ).toThrowErrorMatchingInlineSnapshot(`
```

Comment:
nit: is there a way to just check for string 'Expected string, received number' from the thrown error msg?

Reply:
I kinda prefer the use of inline snapshots for errors, makes updating them later easier because the whole thing is automated

Reply:
ohhh wait I just expand this.. yea this is pretty intense...

we could use `toThrowErrorMatchingSnapshot` instead which will put this in a file outside of the tests rather than inline